Components/Metrics/Impact/itkImpactImageToImageMetric.hxx

```diff
@@ -0,0 +1,1065 @@
+    }
+  }
+  movingImagesPatchesJacobians[s] = torch::from_blob(movingImagesPatchJacobians.data(),
+                                                     { static_cast<int64_t>(patchIndex.size()), MovingImageDimension },
+                                                     torch::kFloat)
+                                      .clone();
+  return torch::from_blob(movingImagesPatchValues.data(), { torch::IntArrayRef(patchSize) }, torch::kFloat)
+    .unsqueeze(0)
+    .clone();
+} // end EvaluateMovingPatchValueAndDerivative
+
+
+/**
+ * ******************* ComputeValue *******************
+ */
+template <typename TFixedImage, typename TMovingImage>
+unsigned int
+ImpactImageToImageMetric<TFixedImage, TMovingImage>::ComputeValue(
+  const std::vector<FixedImagePointType> &         fixedPointsTmp,
+  std::vector<std::unique_ptr<ImpactLoss::Loss>> & losses) const
+{
+  std::vector<std::vector<std::vector<std::vector<float>>>> patchIndex(this->GetFixedModelsConfiguration().size());
+  std::vector<FixedImagePointType>                          fixedPoints =
+    this->GeneratePatchIndex<FixedImagePointType>(this->GetFixedModelsConfiguration(), fixedPointsTmp, patchIndex);
+  if (fixedPoints.empty())
+  {
+    return 0;
+  }
+  unsigned int               nb_sample = fixedPoints.size();
+  std::vector<torch::Tensor> fixedOutputsTensor, movingOutputsTensor;
+
+  std::vector<torch::Tensor> subsetsOfFeatures(this->features_indexes.size());
+
+  for (size_t i = 0; i < this->features_indexes.size(); ++i)
+  {
+    std::vector<int> features_index = this->features_indexes[i];
+    std::shuffle(features_index.begin(), features_index.end(), this->g);
+    std::vector<int> subsetOfFeatures(features_index.begin(), features_index.begin() + this->GetSubsetFeatures()[i]);
+    subsetsOfFeatures[i] = torch::tensor(subsetOfFeatures, torch::kInt64).to(this->GetGPU());
+  }
+
+  fixedOutputsTensor = ImpactTensorUtils::GenerateOutputs<ModelConfiguration, FixedImagePointType>(
+    this->GetFixedModelsConfiguration(),
+    fixedPoints,
+    patchIndex,
+    subsetsOfFeatures,
+    this->GetGPU(),
+    std::function<typename torch::Tensor(
+      const FixedImagePointType &, const std::vector<std::vector<float>> &, const std::vector<int64_t> &)>(
+      [this](const FixedImagePointType &             fixedImageCenterCoordinateLoc,
+             const std::vector<std::vector<float>> & patchIndexLoc,
+             const std::vector<int64_t> &            patchSizeLoc) {
+        return this->EvaluateFixedImagesPatchValue(fixedImageCenterCoordinateLoc, patchIndexLoc, patchSizeLoc);
+      }));
+
+  movingOutputsTensor = ImpactTensorUtils::GenerateOutputs<ModelConfiguration, MovingImagePointType>(
+    this->GetMovingModelsConfiguration(),
+    fixedPoints,
+    patchIndex,
+    subsetsOfFeatures,
+    this->GetGPU(),
+    std::function<typename torch::Tensor(
+      const MovingImagePointType &, const std::vector<std::vector<float>> &, const std::vector<long> &)>(
```

Comment:
@vboussot Just one more `std::vector<long>` ==>  `std::vector<int64_t>`, please 🙏  

Visual C++ says:

`6>D:\Src\ImpactElastix\Components\Metrics\Impact\itkImpactImageToImageMetric.hxx(538,105): error C2440: '<function-style-cast>': cannot convert from 'itk::ImpactImageToImageMetric<itk::Image<float,2>,itk::Image<float,2>>::ComputeValue::<lambda_975d5ea3e56cb8d20f9929b544ff2044>' to 'std::function<at::Tensor (const itk::Point<double,2> &,const std::vector<std::vector<float,std::allocator<float>>,std::allocator<std::vector<float,std::allocator<float>>>> &,const std::vector<long,std::allocator<long>> &)>'
`